pkg/services/authn/clients/proxy.go

```diff
@@ -133,8 +139,9 @@ func (c *Proxy) Hook(ctx context.Context, identity *authn.Identity, r *authn.Req
 	}
 
 	c.log.FromContext(ctx).Debug("Cache proxy user", "userId", id)
-	if err := c.cache.Set(ctx, identity.ClientParams.CacheAuthProxyKey, id, time.Duration(c.cfg.AuthProxySyncTTL)*time.Minute); err != nil {
-		c.log.FromContext(ctx).Warn("Failed to cache proxy user", "error", err, "userId", id)
+	bytes := strconv.FormatInt(id, 10)
+	if err := c.cache.SetByteArray(ctx, identity.ClientParams.CacheAuthProxyKey, []byte(bytes), time.Duration(c.cfg.AuthProxySyncTTL)*time.Minute); err != nil {
```

Comment:
Nit: Instead of converting the id to a string and then a byte slice you could use binary.LittleEndian

e.g
```go
buff := make([]byte, 8)
binary.LittleEndian.PutUint64(buff, uint64(id))
```

And for converting it back
```go
uid := int64(binary.LittleEndian.Uint64(entry))
```

Reply:
TIL! :)

Reply:
@kalleep mind giving my chance a review?

https://github.com/grafana/grafana/pull/63525/commits/e4dd5a01789efd30b20b01b5c973b28a4592a838